Realm/Realm/Handles/CollectionHandleBase.cs

```diff
@@ -24,13 +24,9 @@ namespace Realms
 {
     internal abstract class CollectionHandleBase : NotifiableObjectHandleBase
     {
-        protected delegate IntPtr SnapshotDelegate(out NativeException ex);
-
         public abstract bool IsValid { get; }
 
-        public bool CanSnapshot => SnapshotCore != null;
-
-        protected virtual SnapshotDelegate SnapshotCore => null;
```

Comment:
This refactors `SnaphotCore` so that it's not a virtual property but rather a method being called by `Snapshot`. Addresses https://github.com/realm/realm-dotnet/security/code-scanning/606?query=ref%3Arefs%2Fheads%2Fmaster